server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/api/posttask/PostProjectAnalysisTasksExecutor.java

```diff
@@ -147,13 +153,16 @@ private QualityGateImpl createQualityGate(QualityGateHolder qualityGateHolder) {
     private final CeTask ceTask;
     private final Project project;
     private final Date date;
+    private final ScannerContext scannerContext;
     @CheckForNull
     private final QualityGate qualityGate;
 
-    private ProjectAnalysis(CeTask ceTask, Project project, Date date, @Nullable QualityGate qualityGate) {
+    private ProjectAnalysis(CeTask ceTask, Project project, Date date,
+      ScannerContext scannerContext, @Nullable QualityGate qualityGate) {
       this.ceTask = requireNonNull(ceTask, "ceTask can not be null");
       this.project = requireNonNull(project, "project can not be null");
       this.date = requireNonNull(date, "date can not be null");
+      this.scannerContext = requireNonNull(scannerContext, "scannerContext can not be null");
```

Comment:
@simonbrandhof , since I'm trying myself to stop doing it, I'm passing to you this comment initially from @teryk  and backed up by @ganncamp : this should be `scannerContext can't be null` in American English